core/pubnub_alloc_std.c

```diff
@@ -135,12 +135,18 @@ void pballoc_free_at_last(pubnub_t* pb)
 
 int pubnub_free(pubnub_t* pb)
 {
-    int result = -1;
-
     PUBNUB_ASSERT(pb_valid_ctx_ptr(pb));
 
     PUBNUB_LOG_TRACE("pubnub_free(%p)\n", pb);
 
+    if (pb->state == PBS_NULL) {
+        PUBNUB_LOG_TRACE("pubnub_free(%p) not initialized - freeing...\n", pb);
+        remove_allocated(pb);
+        free(pb);
```

Comment:
Wouldn't there be over release with `remove_allocated`?

Reply:
As i understood -`remove_allocated` only removes concrete instance from the memory. Nothing more.

Reply:
Also it frees only the whole list of contextes if there is nothing left.